.vscode/settings.json

```diff
@@ -1,5 +1,8 @@
 {
   "editor.formatOnSave": true,
   "editor.defaultFormatter": "esbenp.prettier-vscode",
-  "typescript.tsdk": "node_modules/typescript/lib"
+  "typescript.tsdk": "node_modules/typescript/lib",
+  "[typescript]": {
+    "editor.defaultFormatter": "esbenp.prettier-vscode"
+  }
```

Comment:
why the `editor.defaultFormatter` is set in typescript specific rule, when at line 3 it is used for all languages which include typescript. isn't it redundant?

Reply:
You'd think so, but my VSCode instance refuses to recognize the formatter unless it's specifically set like this.